Libs/GObject/Classes/Object.TypeHelper.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace GObject
+{
+    public partial class Object
+    {
+        protected static class TypeHelper
+        {
+            #region Methods
+
+            public static IntPtr GetClassPointer(Type type)
+            {
+                IntPtr ptr = TypeClass.Native.peek(type.Value);
+
+                if (ptr == IntPtr.Zero)
+                    ptr = TypeClass.Native.@ref(type.Value);
+
+                return ptr;
+            }
+            
+            internal static ClassInitFunc GetClassInitFunc(IReflect type) => (gClass, classData) =>
+            {
+                Type gtype = GetGTypeFromTypeClass(gClass);
```

Comment:
In the long term, we might want to look at passing the actual corresponding type class rather than the gtype. For now, this looks good to me :+1: